### DISCUSSION TITLE
Wireguard not working at all or intermittently

### DISCUSSION THREAD

**Discussion by ghazi98** [angristan/wireguard-install#209] [Help]
Hi everyone, a noob here trying to get wireguard working so i can implement it in a corporate environment. I have an ubuntu server running on VM, i also have two windows clients also running on VM. After my setup, i can sometimes get one client to have internet access but when i add another client, there is no internet access. No handshake response. Any reason why? one thing ive noticed is allowed IP's for all clients is the same, 0.0.0.0/0. Will this cause any issues? thanks

Heres a link to my issue:
https://www.reddit.com/r/WireGuard/comments/lrpp7o/wireguard_not_working_with_mulitple_clients/

**Comment by PeerCohere** [comment]
Are all these vms running on the same host? If so that's your problem

**Comment by ghazi98** [comment]
Hi there, yes it’s running on the same computer but I don’t see how this could cause the issue?

**Comment by PeerCohere** [comment]
It's a problem because virtualization uses virtual network drivers. Vmware
and virtual box do anyway.

Two virtual network drivers (wire guard and vm) often don't play well
together out of the box.

I would bet money on this being your problem. You should try Googling the
name of your vm software + wire guard problems. Hopefully someone else has
figured it out.

Not an expert on this just speaking from experience with similar issues and
software.
